fxa-auth-db-server/lib/bufferize.js

```diff
@@ -15,22 +15,38 @@ function unbuffer(object) {
   return object
 }
 
-function bufferize(object, ignore) {
+function bufferize(object, onlyTheseKeys) {
   var keys = Object.keys(object)
+  if (onlyTheseKeys) {
+    keys = keys.filter(key => onlyTheseKeys.has(key))
+  }
   for (var i = 0; i < keys.length; i++) {
     var key = keys[i]
     var value = object[key]
-    if (ignore.indexOf(key) === -1 && typeof value === 'string' && HEX_STRING.test(value)) {
-      object[key] = Buffer(value, 'hex')
+    // Don't convert things with no value, but we still want
+    // to bufferize falsy things like the empty string.
+    if (typeof value !== 'undefined' && value !== null) {
+      if (typeof value !== 'string' || ! HEX_STRING.test(value)) {
+        throw new Error('Invalid hex data for ' + key + ': "' + value + '"')
+      }
+      object[key] = Buffer.from(value, 'hex')
```

Comment:
Fun fact: `Buffer.from(foo, 'hex')` will not throw if you give it invalid hex data:

```
$ node
> Buffer.from('this is clearly not hex data', 'hex')
<Buffer >
> 
> Buffer.from('it only fails if string length is odd', 'hex')
TypeError: Invalid hex string
    at TypeError (native)
    at Buffer.write (buffer.js:631:21)
    at fromString (buffer.js:173:26)
    at Function.Buffer.from (buffer.js:81:12)
    at repl:1:8
    at REPLServer.defaultEval (repl.js:262:27)
    at bound (domain.js:287:14)
    at REPLServer.runBound [as eval] (domain.js:300:12)
    at REPLServer.<anonymous> (repl.js:431:12)
    at emitOne (events.js:82:20)
> 
```